qiskit_ibm_runtime/ibm_runtime_service.py

```diff
@@ -729,6 +730,35 @@ def program(self, program_id: str, refresh: bool = False) -> RuntimeProgram:
 
         return self._programs[program_id]
 
+    def _retrieve_programs(self, name: str = "") -> Dict[str, RuntimeProgram]:
+        """Make an API call to fetch programs.
+
+        Args:
+            name: Name of the program.
+
+        Returns:
+            A dict of ``RuntimeProgram`` instances, keyed by program name.
+        """
+        programs = {}
+        current_page_limit = 20
+        offset = 0
+        while True:
+            response = self._api_client.list_programs(
+                name=name, limit=current_page_limit, skip=offset
+            )
+            program_page = response.get("programs", [])
+            # count is the total number of programs that would be returned if
+            # there was no limit or skip
+            count = response.get("count", 0)
+            for prog_dict in program_page:
+                program = self._to_program(prog_dict)
+                programs[program.program_id] = program
+            if len(programs) == count:
+                # Stop if there are no more programs returned by the server.
+                break
+            offset += len(program_page)
+        return programs
```

Comment:
Given that this PR highly overlaps with https://github.com/Qiskit/qiskit-ibm-runtime/pull/78 why not combine them into this PR and avoid merge conflict upfront? 

Positive side-effect: makes it easier to review your changes in one effort and allows to see if it's consistently solved.

Reply:
Cool, thanks!